test/app_test.dart

```diff
@@ -62,11 +62,89 @@ Future<void> main([List<String>? args]) async {
     final app = App(configuration);
     final credentials = Credentials.anonymous();
     final user = await app.logIn(credentials);
+    expect(user.state, UserState.loggedIn);
   });
+
   test('Application get all users', () {
     final configuration = AppConfiguration(generateRandomString(10));
     final app = App(configuration);
     var users = app.users;
     expect(users.isEmpty, true);
   });
+
+  baasTest('App log out no current user is no operation and does not crash', (configuration) async {
+    final app = App(configuration);
+    await app.logout();
+  });
+
+  baasTest('App log out user', (configuration) async {
+    final app = App(configuration);
+    final user = await app.logIn(Credentials.emailPassword(testUsername, testPassword));
+
+    expect(user.state, UserState.loggedIn);
+    await app.logout(user);
+    expect(user.state, UserState.loggedOut);
+  });
+
+  baasTest('App remove user', (configuration) async {
+    final app = App(configuration);
+    final user = await app.logIn(Credentials.emailPassword(testUsername, testPassword));
+
+    expect(user.state, UserState.loggedIn);
+    await app.removeUser(user);
+    expect(user.state, UserState.removed);
+  });
+
+  baasTest('App log out anonymous user is marked as removed', (configuration) async {
+    final app = App(configuration);
+    final credentials = Credentials.anonymous();
+    final user = await app.logIn(credentials);
+    expect(user.state, UserState.loggedIn);
+    await app.logout(user);
+    expect(user.state, UserState.removed);
+  });
+
+  baasTest('App remove anonymous user', (configuration) async {
+    final app = App(configuration);
+    final credentials = Credentials.anonymous();
+    final user = await app.logIn(credentials);
+    expect(user.state, UserState.loggedIn);
+    await app.removeUser(user);
+    expect(user.state, UserState.removed);
+  });
+
+  baasTest('App get current user', (configuration) async {
+    final app = App(configuration);
+    final credentials = Credentials.anonymous();
+    expect(app.currentUser, isNull);
+
+    final user = await app.logIn(credentials);
+
+    expect(app.currentUser, isNotNull);
+    expect(app.currentUser!.id, user.id);
+  });
+
+  baasTest('App switch user', (configuration) async {
+    final app = App(configuration);
+    expect(app.currentUser, isNull);
+
+    final user = await app.logIn(Credentials.anonymous());
+    expect(app.currentUser!.id, user.id);
+    
+    final user1 = await app.logIn(Credentials.emailPassword(testUsername, testPassword));
+
+    expect(app.currentUser, user1);
+
+    app.switchUser(user);
+    expect(app.currentUser!.id, user.id);
```

Comment:
Why do we check ids here? Can't we use `expect(app.currentUser, user)` like above?

Reply:
this was written before I added the equals operator

Reply:
will fix